controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -681,11 +677,11 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
 
     tlvRuid->radio_uid() = network_utils::mac_from_string(ruid);
 
-    for (int i = 0; i < radio_basic_caps->maximum_number_of_bsss_supported(); i++) {
-        if (!autoconfig_wsc_add_m2(tlvwscM1)) {
-            LOG(ERROR) << "Failed setting M2 attributes";
-            return false;
-        }
+    // TODO add M2 for each VAP to be configured based on DB configuration set by certification
+    // testbed CAPI command dev_set_config
+    if (!autoconfig_wsc_add_m2(tlvwscM1)) {
+        LOG(ERROR) << "Failed setting M2 attributes";
+        return false;
```

Comment:
I don't see a reason for this change. It will be updated as part of #227, but the loop will most likely stay. We will basically have to iterate over the number of supported BSSes in parallel with the BSSes configured i the database.

Reply:
I don't see anyone assigned to #227 - do you plan to do it?
Regarding the change - when a test starts, the agent does onboarding to the controller. At this point `dev_set_config` was not yet requested from the UCC application, so the correct operation the controller needs to take is to tear down all radios, and for that we don't need more than one M2.
Whoever will implement #227 will update this code section to check if there is a configuration in the DB, and if so - use it instead of tear down both radios.
Also - this is how the reference controller behaves during the tests, so it is easier at least for now to stick to the same behavior so we can easily compare sniffer logs of prplMesh controller VS reference controller.

So bottom line - I think we should keep this change.

Reply:
I will do that in a separate commit or even a separate PR, so marking as resolved for now as I removed this commit from the PR.